ttnn/cpp/ttnn/operations/ccl/all_gather/all_gather.cpp

```diff
@@ -9,19 +9,19 @@
 namespace ttnn::operations::ccl {
 
 ttnn::Tensor ExecuteAllGather::invoke(const ttnn::Tensor& input_tensor,
-    const uint32_t dim,
+    const int16_t gather_dim,
     const uint32_t num_links,
     const std::optional<ttnn::MemoryConfig>& memory_config,
     const std::optional<size_t> num_workers,
     const std::optional<size_t> num_buffers_per_channel,
     const ttnn::ccl::Topology topology) {
     return ttnn::operations::ccl::all_gather(
-        input_tensor, dim, num_links, memory_config, num_workers, num_buffers_per_channel, topology);
+        input_tensor, gather_dim, num_links, memory_config, num_workers, num_buffers_per_channel, topology);
 }
 
 ttnn::Tensor ExecuteAllGather::invoke(
     const ttnn::Tensor& input_tensor,
-    const uint32_t dim,
+    const int16_t gather_dim,
```

Comment:
whats the reason for int16 instead of int32?

Reply:
generally, yes, but the place it's packed in the struct right now, I looks like it will still take up the same amount of space.

See https://godbolt.org/z/jM197hxMM as an example:
```

#include <optional>
#include <array>
#include <iostream>
#include <cstdint>

struct S {
    size_t s;
};

struct A final {
    S s;
    std::int16_t x;
    std::int32_t y;
};
struct B final {
    S s;
    std::int32_t x;
    std::int32_t y;
};


int main() {
    std::cout << "sizeof(A): " << sizeof(A{}) << std::endl;
    std::cout << "sizeof(B): " << sizeof(B{}) << std::endl;
    // std::cout << "sizeof(C): " << sizeof({}) << std::endl;
    return 0;
}
```

>>>
sizeof(A): 16
sizeof(B): 16

Reply:
Thanks got it